lineapy/annotations/external/dask.annotations.yaml

```diff
@@ -0,0 +1,64 @@
+- module: dask.dataframe
+  annotations:
+    - criteria:
+        function_names:
+          - read_sql
+          - read_sql_query
+          - read_sql_table
+      side_effects:
+        - dependency:
+            external_state: db
+    - criteria:
+        function_names:
+          - read_csv
+          - read_table
+          - read_fwf
+          - read_parquet
+          - read_hdf
+          - read_json
+          - read_orc
+      side_effects:
+        - dependency:
+            external_state: file_system
+- module: dask.dataframe
+  annotations:
+    - criteria:
+        class_method_names:
+          - to_sql
+        class_instance: DataFrame
+      side_effects:
+        - mutated_value:
+            external_state: db
+    - criteria:
+        class_method_names:
+          - to_csv
+          - to_json
+          - to_html
+          - to_hdf
+          - to_feather
```

Comment:
This comment applies to all functions of the type dataframe.to_*
I cant find a specific line in documentation but i remember reading it somewhere that all these operations force a call to compute internally. If that is the case all of these probably cause a mutation to the dataframe itself and should be annotated for that. here is one reference I could find in someone's code: https://gist.github.com/CMCDragonkai/ad84a4096fa63bc278648db6a9f22a0a

I will try to work on an example to demonstrate this but if you find a clear reference to this, please add here and update the annotations accordingly.

Reply:
the discussion on this issue: https://github.com/dask/dask/pull/5304 discusses the internal where compute is getting called before to_csv on each partition.

Reply:
After futher discussion, it looks like leaving the annotations as is will not affect linea. we can always revisit if a particular scenario shows up. approving this version as is til then.